woocommerce/woocommerce-ios
# HACK week: Handle multipart form upload with URLSessionNetwork

Part 2 of a POC for HACK week in December 2022.


### Description

This PR continues the work from #8345 to handle multipart form upload with URLSession. 

For the quick implementation, I've copied most of Alamofire's logic of `MultipartFormData` to handle data writing to the local file storage for background uploading. `NetworkError` has also been updated with a new error for the failure of encoding multipart form data.

Lastly, `SessionManager` and `URLSessionNetwork` were updated with the implementation of uploading multipart form data. The data is written into the device's file storage, and a request to upload the file is made with a `URLSession`. 

### Testing instructions

- Log in to the app
- Navigate to the Products tab
- Select a product and upload photos for it using files from the device.
- Make sure that photos are uploaded successfully. 

### Screenshots

N/A

---
